test/units/modules/test_iptables.py

```diff
@@ -1,1169 +1,1299 @@
-            'chain': 'INPUT',
-            '_ansible_check_mode': True,
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy DROP)\n', ''),
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
+    return IPTABLES_CMD
+
+
+@pytest.fixture
+def mock_basic_commands(mocker):
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.get_bin_path",
+        return_value=get_bin_path(),
+    )
+    mocker.patch("ansible.modules.iptables.get_iptables_version", return_value=IPTABLES_VERSION)
+
+
+def test_without_required_parameters():
+    with pytest.raises(SystemExit) as e:
+        set_module_args({})
+        iptables.main()
+
+
+def test_flush_table_without_chain(mocker, mock_basic_commands):
+    """Test flush without chain, flush the table"""
+    set_module_args(
+        {
+            "flush": True,
+        }
+    )
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command", return_value=(0, "", "")
+    )
+    with pytest.raises(SystemExit) as e:
+        iptables.main()
+
+    assert run_command.call_count == 1
+    assert run_command.call_args[0][0][0], IPTABLES_CMD
+    assert run_command.call_args[0][0][1], "-t"
+    assert run_command.call_args[0][0][2], "filter"
+    assert run_command.call_args[0][0][3], "-F"
+
+
+def test_flush_table_check_true(mocker, mock_basic_commands):
+    """Test flush without parameters and check == true"""
+    set_module_args(
+        {
+            "flush": True,
+            "_ansible_check_mode": True,
+        }
+    )
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command", return_value=(0, "", "")
+    )
+    with pytest.raises(SystemExit) as e:
+        iptables.main()
+
+    assert run_command.call_count == 0
+
+
+def test_policy_table(mocker, mock_basic_commands):
+    """Test change policy of a chain"""
+    set_module_args(
+        {
+            "policy": "ACCEPT",
+            "chain": "INPUT",
+        }
+    )
+    commands_results = [(0, "Chain INPUT (policy DROP)\n", ""), (0, "", "")]
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+    with pytest.raises(SystemExit) as e:
+        iptables.main()
+
+    assert run_command.call_count == 2
+    assert run_command.call_args_list[0][0][0] == CONST_INPUT_FILTER
+    assert run_command.call_args_list[1][0][0] == [
```

Comment:
May I suggest labeling the data with meaningful names? Like `first_command_call_subprocess_cmd` and `second_command_call_subprocess_cmd`? Perhaps, there's better names but it's hard to say what these actually mean without this context so I had to guess.
If you're able to reduce such guessing, this would reduce the mental overhead for the readers of your code, making it maintainable and easy to grasp.